module/Core/src/Core/Entity/AttachableEntityInterface.php

```diff
@@ -10,29 +10,27 @@
  */
 namespace Core\Entity;
 
-use Core\Entity\Collection\AttachedEntitiesCollection;
-
 interface AttachableEntityInterface
 {
 
     /**
-     * @param AttachedEntitiesCollection $collection
-     * @throws \LogicException If collection is already set
+     * @param AttachableEntityManager $attachableEntityManager
+     * @throws \LogicException If attachable entity manager is already set
      */
-    public function setAttachedEntitiesCollection(AttachedEntitiesCollection $collection);
+    public function setAttachableEntityManager(AttachableEntityManager $attachableEntityManager);
 
     /**
      * @param IdentifiableEntityInterface $entity
      * @param string $key
      * @return AttachableEntityInterface
      */
-    public function attachEntity(IdentifiableEntityInterface $entity, $key = null);
+    public function setAttachedEntity(IdentifiableEntityInterface $entity, $key = null);
```

Comment:
This method should be called "**add**AttachedEntity". Adding an entity with the same key would replace, which must be stated in the phpdoc.